@here/harp-examples/src/globe.ts

```diff
@@ -0,0 +1,204 @@
+            return this.map.canvas;
+        }
+
+        get enabled() {
+            return this.m_enabled;
+        }
+
+        set enabled(enabled: boolean) {
+            if (this.m_enabled === enabled) {
+                return;
+            }
+
+            this.m_enabled = enabled;
+
+            if (enabled) {
+                this.domElement.addEventListener("mousedown", this.m_mouseDownListener, false);
+                this.domElement.addEventListener("wheel", this.m_mouseWheelListener, false);
+            } else {
+                this.domElement.removeEventListener("mousedown", this.m_mouseDownListener, false);
+                this.domElement.removeEventListener("wheel", this.m_mouseWheelListener, false);
+            }
+        }
+
+        private updateCameraMatrix() {
+            this.map.worldCenter.copy(this.camera.position);
+            mapView.projection.scalePointToSurface(this.map.worldCenter);
+            this.camera.updateMatrixWorld(true);
+
+            this.m_cameraMatrix.multiplyMatrices(
+                this.camera.projectionMatrix,
+                this.camera.matrixWorldInverse
+            );
+
+            this.m_cameraMatrixInverse.getInverse(this.m_cameraMatrix);
+        }
+
+        private mouseDown(event: MouseEvent) {
+            event.preventDefault();
+            event.stopPropagation();
+            this.domElement.addEventListener("mouseup", this.m_mouseUpListener, false);
+            this.domElement.addEventListener("mousemove", this.m_mouseMoveListener, false);
+            this.map.beginAnimation();
+            this.updateCameraMatrix();
+        }
+
+        private mouseUp(event: MouseEvent) {
+            event.preventDefault();
+            event.stopPropagation();
+            this.domElement.removeEventListener("mouseup", this.m_mouseUpListener, false);
+            this.domElement.removeEventListener("mousemove", this.m_mouseMoveListener, false);
+            this.map.endAnimation();
+        }
+
+        private mouseMove(event: MouseEvent) {
+            event.preventDefault();
+            event.stopPropagation();
+            const current = this.pointOnSurface(event.clientX, event.clientY);
+            const previous = this.pointOnSurface(
+                event.clientX - event.movementX / window.devicePixelRatio,
+                event.clientY - event.movementY / window.devicePixelRatio
+            );
+            if (current && previous) {
+                const quaternion = new THREE.Quaternion();
+                quaternion.setFromUnitVectors(previous, current);
+                this.m_transform.copyPosition(this.camera.matrix);
+                quaternion.inverse();
+                this.m_transform.makeRotationFromQuaternion(quaternion);
+                this.camera.applyMatrix(this.m_transform);
+                this.updateCameraMatrix();
+            }
+        }
+
+        private mouseWheel(event: WheelEvent) {
+            event.preventDefault();
```

Comment:
we have a harp-map-controls folder, would it make sense to put this logic there?

Reply:
Yes, the code of GlobeControls must be integrated with MapControls.